libpseudodev/pseudodev.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Phoenix-RTOS
+ *
+ * pseudo devices
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/msg.h>
+#include <sys/file.h>
+#include <sys/time.h>
+#include "pseudodev.h"
+
+
+static struct {
+	unsigned int randSeed;
+} pseudo_common;
+
+
+static int pseudo_open(int id, int flags)
+{
+	(void)flags;
+
+	switch (id) {
+		case pseudo_idNull:
+		case pseudo_idZero:
+		case pseudo_idFull:
+		case pseudo_idRandom:
+			return EOK;
+
+		default:
+			break;
+	}
+
+	return -ENODEV;
+}
+
+
+static int pseudo_close(int id)
+{
+	switch (id) {
+		case pseudo_idNull:
+		case pseudo_idZero:
+		case pseudo_idFull:
+		case pseudo_idRandom:
+			return EOK;
+
+		default:
+			break;
+	}
+
+	return -ENODEV;
+}
+
+
+static ssize_t pseudo_read(int id, void *buf, size_t count)
+{
+	switch (id) {
+		case pseudo_idNull:
+			return (ssize_t)0;
+
+		case pseudo_idZero:
+		case pseudo_idFull:
+			memset(buf, 0, count);
+			return (ssize_t)count;
+
+		case pseudo_idRandom:
+			for (size_t ofs = 0; ofs < count; ofs += sizeof(pseudo_common.randSeed)) {
+				rand_r(&pseudo_common.randSeed);
```

Comment:
Nitpick - return value should be used, not seed

Reply:
Done..